File: src/frontend/src/types/pregunta.ts
export enum Tema {
  GENERAL = 'GENERAL',
  TEORIA = 'TEORIA',
  PRACTICA = 'PRACTICA',
  DISENO = 'DISENO',
  IMPLEMENTACION = 'IMPLEMENTACION'
}

export enum Dificultad {
  FACIL = 'FACIL',
  MEDIO = 'MEDIO',
  DIFICIL = 'DIFICIL'
}

export interface Respuesta {
  id?: number;
  opcion: string;
  esCorrecta: boolean;
}

export interface Pregunta {
  id: number;
  enunciado: string;
  tema: string;
  dificultad: Dificultad;
  asignaturaId: number;
  respuestas: Respuesta[];
}

File: src/backend/exam-generator/src/main/java/com/martin/exam_generator/entities/Pregunta.java
package com.martin.exam_generator.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "preguntas")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Pregunta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String enunciado;

    @Column(nullable = false)
    private String tema;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Dificultad dificultad;

    @Column(nullable = false)
    private Boolean habilitada = true;

    @Column(nullable = false)
    private Long docenteId;

    @Column(nullable = false)
    private Long asignaturaId;

    @OneToMany(mappedBy = "pregunta", cascade = CascadeType.ALL, orphanRemoval = true)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private List<Respuesta> respuestas = new ArrayList<>();

    public enum Dificultad {
        FACIL,
        MEDIO,
        DIFICIL
    }
}
